appium/appium-inspector
# ci: update electron-builder and Azure pipeline configurations

This PR updates the configuration for `electron-builder` (which generates OS-specific application packages) and the Azure pipeline that is triggered during the release process (which is where `electron-builder` is run):
* Update Azure VMs to latest OS images
* Bump NodeTool@0 & Node 16 to UseNode@1 & Node 20
* Move `electron-builder` config to a separate file for clarity
* Generate x64 and arm64 builds for all 3 OSes
  * Remove Windows ia32 builds
  * Split macOS universal artifacts into x64 and arm64 artifacts (smaller download)
* Use common naming format for all artifacts
  * Note that Linux x64 AppImage is for some reason generated with the `x84_64` suffix instead of `x64`
* Add `tar.gz` artifacts for Linux
* Add entitlements file for macOS (likely only relevant after updating to Electron 20+)

As a result of these changes, I believe the number of generated artifacts will now increase to 26.
Resolves #1400.
